src/index.js

```diff
@@ -1,17 +1,67 @@
 const isDefined = (o) => ![null, undefined].includes(o);
 
-const getFarContext2d = (canvas, { x = 0, y = 0, scale = 1 } = {}) => {
-  const d = { x, y, scale };
+const getFarContext2d = (
+  canvas,
+  { x = 0, y = 0, scale = 1, rotation = { x: 0, y: 0, angle: 0 } } = {}
+) => {
+  if (![0, Math.PI].includes(rotation.angle)) {
+    throw new Error("Only 0 and PI rotation angles are supported");
+  }
+
   const _context = canvas.getContext("2d");
 
+  const d = { x, y: y * Math.cos(rotation.angle), scale, rotation };
+
   const s = {
-    x: (x) => d.scale * (x + d.x),
-    y: (y) => d.scale * (y + d.y),
-    distance: (distance) => distance * d.scale,
+    x: (x) => {
+      // First, translate, then rotate
+      const translatedX = x + d.x;
+      const translatedY = d.y; // y-coordinate remains the same for calculating x
+      // Apply rotation
+      const rotatedX =
+        Math.cos(d.rotation.angle) * (translatedX - d.rotation.x) -
+        Math.sin(d.rotation.angle) * (translatedY - d.rotation.y) +
+        d.rotation.x;
+      // Finally, apply scaling
+      return d.scale * rotatedX;
+    },
+    y: (y) => {
+      // First, translate, then rotate
+      const translatedX = d.x; // x-coordinate remains the same for calculating y
+      const translatedY = y + d.y;
+      // Apply rotation
+      const rotatedY =
+        Math.sin(d.rotation.angle) * (translatedX - d.rotation.x) +
+        Math.cos(d.rotation.angle) * (translatedY - d.rotation.y) +
+        d.rotation.y;
+      // Finally, apply scaling
+      return d.scale * rotatedY;
+    },
+    distance: (distance) => distance * d.scale, // Scale distances
     inv: {
-      x: (x) => x / d.scale - d.x,
-      y: (y) => y / d.scale - d.y,
-      distance: (distance) => distance / d.scale,
+      x: (x) => {
+        // First, undo scaling
```

Comment:
comments are superfluous here, the variable name and it's context should be more then enough